### PR TITLE
feat(sign-in-with-google): create the Google SignIn feature #11

### PR DESCRIPTION
**Pull request by ALVINdimpos**
What does this PR do?

       This PR adds the ability for users to log in using their Google account.

    Description of Task to be completed

        Implement Google OAuth2 authentication in the backend
        Create a new endpoint for Google login and handle the login process
        Store the user's Google ID and access token securely in the database
        Ensure the user is redirected to the correct page after logging in
        Implement the ability for users to log out of their Google account and log in again using the same credentials
        Document the implementation following best practices for security and maintainability

    How should this be manually tested?

         npm install
         npm run dev
        then follow this route http://localhost:3000/api/auth/google/callback

    Any background context you want to provide?

        This feature was requested by users who prefer to log in using their Google accounts rather than creating a new                                         email/password account.
        What are the relevant pivotal tracker stories?

        CardId #11: Add Google login feature
        CardLink:https://trello.com/c/FQfu8x6Y/11-wip-sign-in-with-social-accounts

    Screenshots (if appropriate)

    N/A (or add relevant screenshots if appropriate)

### CODE REVIEW

**Comment by ALVINdimpos** [comment]
[[WIP] Sign In with Social Accounts](https://trello.com/c/FQfu8x6Y/11-wip-sign-in-with-social-accounts)

**Review by Muhet — COMMENTED** [review]
keep it up man, good work

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -24,17 +24,4 @@ jobs:
...
-      - name: Wait for Render Deployment
-        uses: bounceapp/render-action@0.6.0
-        with:
-          render-token: ${{ secrets.RENDER_TOKEN }}
-          github-token: ${{ secrets.TOKEN }}
-          service-id: srv-cgksc9seoogkndl3t3qg
-          retries: 20
-          wait: 16000
-          sleep: 30000
+
```
Comment (chanceDira) [inline .github/workflows/node.js.yml:27]: Why deploy job is removed?

File: src/controller/authController.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    "1h"
+  );
+  return res.redirect(`/api/callback?key=${userToken}`);
+};
+// get the user from the database
+
+export const GetUsers = async (req, res) => {
+  try {
+    const users = await db.User.findAll();
+    return res.status(200).json({ users });
```
Comment (chanceDira) [inline src/controller/authController.js]: Let's use Jsend format to structure our responses.

Comment (chanceDira) [inline src/controller/authController.js]: Let's use Jsend format to structure our responses.

**Review by chanceDira — CHANGES_REQUESTED** [review]
Nice work @ALVINdimpos, I just added some feedback for you.

Also, let's use this PR example for team consistency 
https://github.com/atlp-rwanda/engineering-playbook/wiki/3.-Pull-Request

File: src/controller/authController.js (hunk 2 above)
Comment (ALVINdimpos) [reply src/controller/authController.js]: done!

Comment (ALVINdimpos) [reply src/controller/authController.js]: done!

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (ALVINdimpos) [reply .github/workflows/node.js.yml:27]: i have remove it becouse it require everyone to deploy their own repo in order to pass check and also to set their own RENDER_TOKEN

**Review by bryan936M — APPROVED** [review]
Keep up the good work.

**Review by yobuir — APPROVED** [review]
Great job @ALVINdimpos

**Review by faid-terence — APPROVED** [review]
Good job @ALVINdimpos

**Review by NTElissa — APPROVED** [review]
LGTM

**Review by vanessa-abewe — APPROVED** [review]

**Comment by ALVINdimpos** [comment]
> LGTM

what do u mean sir?

**Review by Muhet — APPROVED** [review]
well done broh

**Review by chanceDira — APPROVED** [review]
Good work @ALVINdimpos

**Pull request merged**